pulp_container/tests/functional/api/test_push_signatures.py

```diff
@@ -0,0 +1,155 @@
+"""Tests that verify that an image signature can be pushed to Pulp."""
+import gnupg
+import base64
+import json
+import requests
+import shutil
+import subprocess
+import pytest
+import tempfile
+
+from pulp_container.tests.functional.constants import REGISTRY_V2_REPO_PULP
+from pulp_container.constants import SIGNATURE_TYPE
+
+KEY = "https://raw.githubusercontent.com/pulp/pulp-fixtures/master/common/GPG-PRIVATE-KEY-pulp-qe"
+
+
+@pytest.fixture(scope="session")
+def tempdir():
+    """A temporary directory."""
+    directory = tempfile.mkdtemp()
+    yield directory
+    shutil.rmtree(directory)
+
+
+@pytest.fixture(scope="session")
+def gpg_metadata(tempdir):
+    """A fixture for a GPG instance and related metadata."""
+    private_key = requests.get(KEY).text
+
+    gpg = gnupg.GPG(gnupghome=tempdir)
+
+    gpg.import_keys(private_key)
+
+    fingerprint = gpg.list_keys()[0]["fingerprint"]
+    keyid = gpg.list_keys()[0]["keyid"]
+
+    gpg.trust_keys(fingerprint, "TRUST_ULTIMATE")
+
+    return gpg, (fingerprint, keyid)
```

Comment:
I think this is a result of some evolution, right? Maybe just return
```suggestion
    return gpg, fingerprint, keyid
```

Or we create a named tuple for this?
But we can also take it as is and think about reusability later.

Reply:
I really like the key handling as a fixture, and eventually i want to see this move to pulp_smash in a way that can be used by pulp_ansible tests too.